ui/round/src/view/replay.ts

```diff
@@ -129,23 +135,37 @@ export function analysisButton(ctrl: RoundController): LooseVNode {
   );
 }
 
-function renderButtons(ctrl: RoundController) {
-  const d = ctrl.data,
-    firstPly = util.firstPly(d),
-    lastPly = util.lastPly(d);
+const goThroughMoves = (ctrl: RoundController) => {
+  return function (e: MouseEvent) {
+    e.preventDefault();
+    let timeoutId: number | undefined = undefined;
+    const clearMovesTimeout = () => {
+      clearTimeout(timeoutId);
+    };
+    const targetPly = (e: MouseEvent) => parseInt((e.target as HTMLElement).getAttribute('data-ply') || '');
+    const delay = showMovesDecreasingDelay();
+    document.addEventListener(e.type === 'touchstart' ? 'touchend' : 'mouseup', () => clearMovesTimeout(), {
+      once: true,
+    });
```

Comment:
the code is a bit tricky, and duplicated with the round repeater. Can they be refactored?